src/isomorphic/classic/class/ReactClass.js

```diff
@@ -728,6 +728,11 @@ var ReactClassMixin = {
    * type signature and the only use case for this, is to avoid that.
    */
   replaceState: function(newState, callback) {
+    if (this.updater.isFiberUpdater) {
```

Comment:
Took me a while to figure out. If the updater schedules them separately, and the update is synchronous, then the work is already done by the time the callback is scheduled, so the callback doesn't get called until the next flush. This is what I was trying to describe on Friday that is similar to the `componentWillMount` bug we found.

In incremental mode, the flow goes like this:

- (in updater) Schedule update
- (in updater) Schedule callback
- Perform work

In sync mode, it goes like this

- (in updater) Schedule update
- Because the update is synchronous, the work is performed immediately
- (in updater) Schedule callback

I'm actually not sure how Stack deals with this, come to think of it, but this seemed like the most straightforward solution.